src/ucp/core/ucp_am.c

```diff
@@ -21,6 +21,37 @@
 #include <ucp/dt/dt.inl>
 
 
+ucs_status_t ucp_am_init(ucp_worker_h worker)
+{
+    ucp_am_cb_data_t *am;
+
+    if (!(worker->context->config.features & UCP_FEATURE_AM)) {
+        return UCS_OK;
+    }
+
+    am = &worker->am;
+
+    am->cbs_array_len = UCP_AM_CB_BLOCK_SIZE;
+    am->cbs           = ucs_calloc(am->cbs_array_len, sizeof(ucp_am_entry_t),
```

Comment:
why is it better to allocate the array up-fron, vs. allocating on demand?
allocating on demand will be less code

Reply:
not sure, just thought it would be better here.
I changed my mind now, will revert